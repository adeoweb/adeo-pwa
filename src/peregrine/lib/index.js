// components
export { default as PeregrineContextProvider } from './PeregrineContextProvider';
