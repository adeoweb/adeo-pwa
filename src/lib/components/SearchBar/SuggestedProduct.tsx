import React, { FunctionComponent, useCallback } from 'react';
import { useTranslation } from 'react-i18next';

import Price from '@magento/peregrine/lib/Price';

import Image from 'src/lib/components/Image';
import RichText from 'src/lib/components/RichText';
import { Link } from 'src/lib/drivers';
import { TProductSearch } from 'src/lib/types/ProductSearch';
import getItemUrl from 'src/lib/util/getItemUrl';

export type TSuggestedProductProps = {
    onNavigate: () => void;
} & TProductSearch;

const IMAGE_WIDTH = 60;

const SuggestedProduct: FunctionComponent<TSuggestedProductProps> = product => {
    const { t } = useTranslation('product');
    const {
        small_image: smallImage,
        name,
        onNavigate,
        price_range: priceRange,
        short_description: shortDescription
    } = product;

    const handleClick = useCallback(() => {
        if (typeof onNavigate === 'function') {
            onNavigate();
        }
    }, [onNavigate]);

    const imageResource =
        smallImage && smallImage.url ? smallImage.url : undefined;

    const uri = getItemUrl(product);

    let price: React.ReactNode | null = null;
    if (
        priceRange?.maximum_price?.final_price?.currency &&
        priceRange?.maximum_price?.final_price?.value
    ) {
        const { currency, value } = priceRange.maximum_price.final_price;
        price = <Price currencyCode={currency} value={value} />;
    }

    return (
        <Link to={uri} onClick={handleClick}>
            <div className="suggestion-left">
                <Image
                    alt={name || ''}
                    resource={imageResource}
                    width={IMAGE_WIDTH}
                />
            </div>
            <div className="suggestion-right">
                <div className="product-line product-name">{name}</div>
                <div className="product-line product-price">
                    {t('Price')}: {price}
                </div>
                <div className="product-des">
                    <p className="short-des">
                        {shortDescription && shortDescription.html && (
                            <RichText content={shortDescription.html} />
                        )}
                    </p>
                </div>
            </div>
        </Link>
    );
};

export default SuggestedProduct;
