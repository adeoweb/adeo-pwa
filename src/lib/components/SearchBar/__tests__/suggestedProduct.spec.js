import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import SuggestedProduct from '../SuggestedProduct';
import { IntlProvider } from 'react-intl';

jest.mock('../../../classify');
jest.mock('src/lib/drivers', () => ({
    Link: ({ children }) => children,
    resourceUrl: jest.fn(src => src),
    useHistory: jest.fn()
}));

const defaultProps = {
    onNavigate: jest.fn(),
    url_key: 'urlKey',
    small_image: {
        url: 'https://small.image/url'
    },
    name: 'product name',
    price_range: {
        maximum_price: {
            final_price: {
                value: 39,
                currency: 'EUR'
            }
        }
    },
    short_description: {
        html: 'short desc'
    }
};

test('renders a suggestedProduct component', () => {
    const tree = createTestInstance(
        <IntlProvider locale="en">
            <SuggestedProduct {...defaultProps} />
        </IntlProvider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
