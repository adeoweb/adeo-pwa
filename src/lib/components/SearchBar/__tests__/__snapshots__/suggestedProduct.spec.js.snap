// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a suggestedProduct component 1`] = `
Array [
  <div
    className="suggestion-left"
  >
    <div
      className="undefined undefined"
    >
      <img
        alt="product name"
        className="undefined undefined"
        loading="eager"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
        style={
          Object {
            "height": "auto",
            "width": "60px",
          }
        }
        width={60}
      />
      <img
        alt="product name"
        className="undefined undefined"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="60px"
        src="https://small.image/url"
        srcSet="https://small.image/url 40w,
https://small.image/url 80w,
https://small.image/url 160w,
https://small.image/url 320w,
https://small.image/url 640w,
https://small.image/url 960w,
https://small.image/url 1280w,
https://small.image/url 1600w,
https://small.image/url 2560w"
        style={
          Object {
            "height": "auto",
            "width": "60px",
          }
        }
        width={60}
      />
    </div>
  </div>,
  <div
    className="suggestion-right"
  >
    <div
      className="product-line product-name"
    >
      product name
    </div>
    <div
      className="product-line product-price"
    >
      Price
      : 
      <span>
        â‚¬
      </span>
      <span>
        39
      </span>
      <span>
        .
      </span>
      <span>
        00
      </span>
    </div>
    <div
      className="product-des"
    >
      <p
        className="short-des"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "short desc",
            }
          }
          onClick={[Function]}
        />
      </p>
    </div>
  </div>,
]
`;
